parlai/scripts/self_chat.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""
+Allows a model to self-chat on a given task.
+"""
+from parlai.core.params import ParlaiParser
+from parlai.core.agents import create_agent
+from parlai.core.worlds import create_task
+
+import random
+
+
+def setup_args(parser=None):
+    if parser is None:
+        parser = ParlaiParser(True, True, 'Self chat with a model')
+    parser.add_argument('-d', '--display-examples', type='bool', default=True)
+    parser.add_argument(
+        '--display-prettify',
+        type='bool',
+        default=False,
+        help='Set to use a prettytable when displaying '
+        'examples with text candidates',
+    )
+    parser.add_argument(
+        '--display-ignore-fields',
+        type=str,
+        default='label_candidates,text_candidates',
+        help='Do not display these fields',
+    )
+    parser.add_argument(
+        '-it',
+        '--interactive-task',
+        type='bool',
+        default=True,
+        help='Create interactive version of task',
+    )
+    parser.set_defaults(interactive_mode=True, task='self_chat')
+    return parser
+
+
+def self_chat(opt, print_parser=None):
+    if print_parser is not None:
+        if print_parser is True and isinstance(opt, ParlaiParser):
+            print_parser = opt
+        elif print_parser is False:
+            print_parser = None
+    if isinstance(opt, ParlaiParser):
+        print('[ Deprecated Warning: self_chat should be passed opt not Parser ]')
+        opt = opt.parse_args()
+
+    # Create models
+    agent1 = create_agent(opt, requireModelExists=True)
+    agent2 = agent1.clone()
+    if hasattr(agent2, 'id'):
+        agent2.id = agent2.id + "2"
+
+    world = create_task(opt, [agent1, agent2])
+
+    if print_parser:
+        # Show arguments after loading model
+        print_parser.opt = agent1.opt
+        print_parser.print_args()
+
+    # Show some example dialogs:
+    while True:
+        world.parley()
+        if opt.get('display_examples'):
+            print("---")
+            print(world.display())
+        if world.epoch_done():
+            print("EPOCH DONE")
+            break
+
+
+if __name__ == '__main__':
+    random.seed(42)
```

Comment:
Can we make seed an argument? I find it infuriating to not be able to easily switch up the personas etc.